ts/Core/Series/DataLabel.ts

```diff
@@ -814,10 +814,10 @@ namespace DataLabel {
                 // Destroy and remove the inactive ones
                 let j = dataLabels.length;
                 while (j--) {
-                    if (dataLabels[j].isActive) {
+                    if (dataLabels[j]?.isActive) {
```

Comment:
According to TypeScript, `dataLabels` is `(SVGElement | SVGLabel)[]`. So somewhere the typing have failed us by allowing an undefined value in the array. Can we dig a bit deeper and find out why/where that happens? It may prevent future issues from the same cause.

Reply:
The problem here is that we map through the [pointOptions](https://github.com/highcharts/highcharts/blob/dcf0ac67b30ed690429aac019c5eceb3a1139aaf/ts/Core/Series/DataLabel.ts#L561). Then, we assign the `dataLabels` to an array by the mapping index, but only when the `dataLabel` is enabled. In effect, when e.g. the first `dataLabel` isn't enabled, and the second one is, we have an empty element inside the array, so the `dataLabels[j].isActive` breaks. On the other hand, as you mentioned, we assume that all elements are defined. 

What's more, I think we need to be aware of two cases, new labels, and the existing ones we want to update. Therefore, we should use the mapping indices we have now to properly add or update `dataLabels`, and create elements even if they aren't enabled. I think we need these only in the scope of mapping, and this [check](https://github.com/highcharts/highcharts/blob/dcf0ac67b30ed690429aac019c5eceb3a1139aaf/ts/Core/Series/DataLabel.ts#L815) can base on formatted (enabled only) `dataLabels`.

Reply:
What if we have only one `dataLabel` setting, and that has `enabled: false`? In that case we don't create an element, right? And for a good reason - turning off the data labels will increase performance. So we need to consider allowing these undefined places in the `dataLabels` array. Like it currently works, just with correct typing?

Reply:
Yes, I absolutely agree to this. So we allow `undefined` only in the local scope.

Now related to your latest commit, I think the `// Write back` block can be simplified. Setting `point.dataLabel` to `point.dataLabels[0]` should be save because it is either undefined or an element. For `point.dataLabels` we should be able to do something like this:

`point.dataLabels = dataLabels.filter(defined)`

Or alternatively do the handling further up, so that `dataLabels` is either undefined or filtered.

Reply:
Yes, I think everything is properly typed now. Thanks for the refactoring suggestion, I pushed the changes.